### ISSUE TITLE
Documentation improvement: What is a gist and what is it used for

### ISSUE THREAD

**Issue by Alestrix** [isaacs/github#1767]
The page about gists at https://help.github.com/en/github/writing-on-github/editing-and-sharing-content-with-gists is missing a fundamental explanation: What is a gist and what is it used for?
After googling around I found good explanations
here: https://stackoverflow.com/questions/6767518/what-is-the-difference-between-github-and-gist and
here: https://robertgreiner.com/using-github-as-a-syntax-highlighter/

In short, they are usually used to share useful code snippets or other little pieces of information. But that definition should be in the help page about gists and not somewhere in the interwebs.

Background: I was recently asked to share some information "with a gist" and the GitHub page didn't help me at all what that guy was actually talking about.
